.github/workflows/update-go-versions.yml

```diff
@@ -0,0 +1,38 @@
+---
+name: Weekly Go Version Check and Metric Generation
+
+on:
+  schedule:
+    - cron: "0 0 * * 0"  # Every Sunday at midnight UTC
```

Comment:
According to [go release docs](https://go.dev/wiki/Go-Release-Cycle), new releases are created in February and August. Maybe we could limit this workflow to run only during those months?